The "fd" argument must be of type number. Received type undefined

Fresh new install of bids-validator. 
Checking a dataset mostly in bids format (maybe missing the dataset_description and participants file).  Data are converted by someone else, so not sure what exactly is the problem. Here is the error (running as root, so no permission problems). Any idea?

```
internal/validators.js:113
    throw err;
    ^

TypeError [ERR_INVALID_ARG_TYPE]: The "fd" argument must be of type number. Received type undefined
    at Object.read (fs.js:448:3)
    at /usr/lib/node_modules/bids-validator/utils/files.js:252:20
    at FSReqWrap.oncomplete (fs.js:145:20)
```

Installing from branch is not working, will need to check later if there is any specific problem with `npm` permissions, unless you know what is going on. Thanks.

```
[root@mri dorian]# npm install -g bids-validator
/usr/bin/bids-validator -> /usr/lib/node_modules/bids-validator/bin/bids-validator
+ bids-validator@0.26.15
added 109 packages from 69 contributors in 12.144s
[root@mri dorian]# npm uninstall -g bids-validator
removed 109 packages in 1.045s
[root@mri dorian]# npm install -g chrisfilo/bids-validator#enh/better_error_reporting
npm ERR! code 128
npm ERR! Command failed: /opt/anaconda2/bin/git clone --depth=1 -q -b enh/better_error_reporting git://github.com/chrisfilo/bids-validator.git /root/.npm/_cacache/tmp/git-clone-91c88a1f
npm ERR! fatal: could not create leading directories of '/root/.npm/_cacache/tmp/git-clone-91c88a1f': Permission denied
npm ERR!

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2018-07-06T20_17_42_987Z-debug.log
[root@mri dorian]#
```

No idea. The installation worked for me just fine.



On Fri, Jul 6, 2018 at 1:19 PM dorianps <notifications@github.com> wrote:

> Installing from branch is not working, will need to check later if there
> is any specific problem with npm permissions, unless you know what is
> going on. Thanks.
>
> [root@mri dorian]# npm install -g bids-validator
> /usr/bin/bids-validator -> /usr/lib/node_modules/bids-validator/bin/bids-validator
> + bids-validator@0.26.15
> added 109 packages from 69 contributors in 12.144s
> [root@mri dorian]# npm uninstall -g bids-validator
> removed 109 packages in 1.045s
> [root@mri dorian]# npm install -g chrisfilo/bids-validator#enh/better_error_reporting
> npm ERR! code 128
> npm ERR! Command failed: /opt/anaconda2/bin/git clone --depth=1 -q -b enh/better_error_reporting git://github.com/chrisfilo/bids-validator.git /root/.npm/_cacache/tmp/git-clone-91c88a1f
> npm ERR! fatal: could not create leading directories of '/root/.npm/_cacache/tmp/git-clone-91c88a1f': Permission denied
> npm ERR!
>
> npm ERR! A complete log of this run can be found in:
> npm ERR!     /root/.npm/_logs/2018-07-06T20_17_42_987Z-debug.log
> [root@mri dorian]#
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/INCF/bids-validator/issues/483#issuecomment-403136758>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAOkpxqe-RDj_DN8_XMcuMLoktN58WkPks5uD8ZogaJpZM4VF33Z>
> .
>

--- a/utils/files.js
+++ b/utils/files.js
@@ -247,14 +247,19 @@ function readNiftiHeader (file, callback) {
                     callback(handleGunzipError(buffer, file));
                 });
 
-            fs.open(file.path, 'r', function(status, fd) {
-                fs.read(fd, buffer, 0, bytesRead, 0, function() {
-                    if (file.name.endsWith('.nii')) {
-                        callback(parseNIfTIHeader(buffer, file));
-                    } else {
-                        decompressStream.write(buffer);
-                    }
-                });
+            fs.open(file.path, 'r', function(err, fd) {
+                if (err){
+                    callback({error: new Issue({code: 44, file: file})});
+                    return;
+                } else {
+                    fs.read(fd, buffer, 0, bytesRead, 0, function () {
+                        if (file.name.endsWith('.nii')) {
+                            callback(parseNIfTIHeader(buffer, file));
+                        } else {
+                            decompressStream.write(buffer);
+                        }
+                    });
+                }
             });
         });
     } else {

